metadata.yaml

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: opensearch-dashboards
+display-name: Opensearch Dashboards
+description: |
+    OpenSearch Dashboard is a community-driven, Apache 2.0-licensed open source 
+    user interface that lets you visualize your OpenSearch data and run and scale
+    your OpenSearch clusters.
+
+summary: The Charmed Opensearch Dashboards operator
+maintainers:
+  - Judit Novak <judit.novak@canonical.com>
+series:
+  - jammy
+
+peers:
+  dashboard_peers:
+    interface: dashboard_peers 
+  restart:
```

Comment:
With the new Charm Structure terminlogy I'm a bit worried to use such a generic name.  We may need it for something else, who knows :-) 

Renamed to `rolling_operations`.

Reply:
yes, you don't need multiple relations - you can use the same `service` relation and use the `override_callback=new_callback` attribute

Reply:
I understand that 'it works'... But are you sure the this is a designated usage...?

It's not recommended by the documentation. I.e. "officially not supported", if I get it right.

What it does, in reality is to alter, "re-use" the relation data each time with a new callback... I'm wondering if we can count on this functionality being supported in a long run... As it seems as a hidden feature rather than a well-established functionality.

I clearly see the advantage of _not_ defining N number of relations. This just doesn't seem to be part of the proposed interface.

Anyway.

If this is what you want to have, I can implement it.

Reply:
Just adding thoughts here as I (for my sins) added the `callback_override`.
Each relation is it's own lock for a specific functionality, and we 'might' want to support more than one function callback in case we have slightly differing implementations. For Kafka, it was 'Disable + re-enable the snap, THEN restart', as opposed to always just 'restarting'. 

If we have every locking function in the same relation, then every function rolls and blocks eachother. Let's say we have two operations that need to run on only 1 unit at a time, for example `restart` and `reshard`.

We may be fine with resharding a unit when another unit restarts, and it's a good idea to not block it within a single relation.

If we need every operation to run one-by-one, then 1 relation.
If we can have them run in parallel, then N relations.

Reply:
For the sake of completeness and correctness - this is charm specific. 
I had forgotten that the dashboards charm is stateless, thus not needing to have a rolling "start" - the rolling need is only for "restart"s - unlike opensearch which needs rolling for both.

Reply:
I see your point... However once we have a load-balanced service, it becomes importnant not to have all nodes going down in the same time resulting in service unavailability... If we can avoid that.

I think --once we have such a nice feature available, fully implemented-- we should just keep using it :-)

I wouldn't like to remove `rolling-ops` from the Dashboard, I think it's a useful feature.

Reply:
I meant the opposite - we should allow all units to go "up" / start (`_on_start` hook) at the same time, without rolling. Rolling ops stays for "restarts"

Reply:
Ah, OK, I see your point, and it's a valid point indeed. 

Let me for a way to do that (here or another PR) -- more news later.

Reply:
I made an attempt to implement this, see latest commit.